src/game.cpp

```diff
@@ -727,17 +727,18 @@ void game::reenter_fullscreen()
     }
 }
 
-/*
- * Initialize more stuff after mapbuffer is loaded.
- */
 void game::setup()
 {
     popup_status( _( "Please wait while the world data loads..." ), _( "Loading core data" ) );
     load_core_data();
 
     load_world_modfiles(world_generator->active_world);
+    MAPBUFFER.init( world_generator->active_world->world_path + "/maps" );
 
-    m =  map( get_option<bool>( "ZLEVELS" ) );
+    m = map( get_option<bool>( "ZLEVELS" ) );
+    if( true ) {
```

Comment:
Why? And why is the `map_memory_ptr` optional (needs checking before accessing it). It should probably be like `game::m`/`game::map_ptr`.

Reply:
It is optional because it may be expensive early on, before I optimize it. I wanted to avoid the situation where a half-implemented feature may fry toasters without the ability to turn it off.

Reply:
Instead of manual checks for pointer null-ness, you can create a map subclass that has little to no contents and returns null-ish results for everything it needs to supply.